File: linguagem-java/emerson_moreira/java/Bubble.java
class Bubble{
  void bubbleSort(int arr[]){
    int aux;
    

    for(int i = 0; i < arr.length; i++){
      for(int j = 0; j < arr.length -1 ; j++){
        if(arr[j] > arr[j+1] ){
          aux = arr[j];
          arr[j] = arr[j+1];
          arr[j+1] = aux;
        }
      }
    }
    printBubble(arr);
  }
  
  void printBubble(int array[]){
    for(int i = 0; i < array.length; i++){
      System.out.print(array[i] + " ");
    }
  }
}

File: liguagem-php/emerson_moreira/php/bubble.php
<?php
class Bubble_Sort{
  function bubbleSort($arr){
    $swapped=true;
    $n=count($arr);
    $temp=null;
    while($swapped){
        $swapped = false;
        for($i=0; $i<$n-1; $i++){
            if( $arr[$i]>$arr[$i+1]){
                $temp=$arr[$i];
                $arr[$i]=$arr[$i+1];
                $arr[$i+1]=$temp;
                $swapped=true;
            }
        }
    }
 
    return $arr;
  }
}

 
